library/network/src/lib/y2firewall/firewalld/zone.rb

```diff
@@ -77,24 +94,20 @@ def full_name
       #
       # @return [Boolean] true if it was modified; false otherwise
       def modified?
-        return true if current_interfaces.sort != interfaces.sort
-        return true if current_services.sort   != services.sort
-        return true if current_protocols.sort  != protocols.sort
-        return true if current_ports.sort      != ports.sort
-        return true if current_sources.sort    != sources.sort
-
-        masquerade? != api.masquerade_enabled?(name)
+        !modified.empty?
       end
 
       # Apply all the changes in firewalld but do not reload it
       def apply_changes!
-        apply_interfaces_changes!
-        apply_services_changes!
-        apply_ports_changes!
-        apply_protocols_changes!
-        apply_sources_changes!
+        return true unless modified?
+
+        apply_relations_changes!
+        if modified.include?(:masquerade)
```

Comment:
do you expect that modified will contain something else than :modified ever?

Reply:
what do you mean?, in modified you will have something like [:services, :interfaces] so the rest of the attributes and relations don't have to be touched avoiding api calls to check what was changed or avoiding calls to change them.